File: src/main/java/org/minturtle/careersupport/user/dto/UserLoginResponse.java
package org.minturtle.careersupport.user.dto;


import lombok.AllArgsConstructor;
import lombok.Getter;
import org.minturtle.careersupport.user.entity.User;

@Getter
@AllArgsConstructor
public class UserLoginResponse {
    private final String nickname;
    private final String token;


    public static UserLoginResponse of(User user, String token){
        return new UserLoginResponse(user.getNickname(), token);
    }
}

File: src/main/java/org/minturtle/careersupport/user/dto/UserLoginResponse.kt
package org.minturtle.careersupport.user.dto

import org.minturtle.careersupport.user.entity.User

data class UserLoginResponse(

    val nickname: String,
    val token: String

){

    companion object{
        @JvmStatic
        fun of(user : User, token: String): UserLoginResponse {
            return UserLoginResponse(
                user.nickname,
                token
            )
        }
    }

}
